File: src/classes/api/req_res/upload-files-request.ts
'use strict';

import { FileData } from '../file-data';
import { GkillAPIRequest } from '../gkill-api-request';
import { FileUploadConflictBehavior } from './file-upload-conflict-behavior';


export class UploadFilesRequest extends GkillAPIRequest {


    files: Array<FileData>;

    target_rep_name: string;

    conflict_behavior: FileUploadConflictBehavior;

    constructor() {
        super()
        this.files = new Array<FileData>()
        this.target_rep_name = ""
        this.conflict_behavior = FileUploadConflictBehavior.rename
    }


}

File: src/app/gkill/api/req_res/upload_files_request.go
package req_res

type UploadFilesRequest struct {
	SessionID string `json:"session_id"`

	Files []*FileData `json:"files"`

	TargetRepName string `json:"target_rep_name"`

	ConflictBehavior FileUploadConflictBehavior `json:"conflict_behavior"`
}
